tools/graphiteindex/src/main/java/com/criteo/biggraphite/graphiteindex/LuceneIndexSearcher.java

```diff
@@ -0,0 +1,249 @@
+import org.apache.cassandra.db.*;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.db.partitions.UnfilteredPartitionIterator;
+import org.apache.cassandra.db.rows.UnfilteredRowIterator;
+import org.apache.cassandra.index.Index;
+import org.apache.cassandra.index.sasi.plan.QueryController;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.IntPoint;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.*;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.automaton.RegExp;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Lucene-based Graphite metrics index.
+ *
+ * TODO(d.forest): provide support for globstar (**)
+ *
+ * TODO(d.forest): provide support for autocompletion
+ */
+public class LuceneIndexSearcher implements Index.Searcher, Closeable
+{
+    private static final Logger logger = LoggerFactory.getLogger(LuceneIndexSearcher.class);
+
+    private final ColumnFamilyStore baseCfs;
+    private final ReadCommand readCommand;
+    private final ColumnDefinition column;
+    private final QueryController queryController;
+
+    /**
+     * TODO(p.boddu): Add support to read from in-memory LuceneIndex. Including ControlledRealTimeReopenThread
+     */
+    public LuceneIndexSearcher(ColumnFamilyStore baseCfs, ColumnDefinition column, ReadCommand readCommand)
+        throws IOException
+    {
+        this.baseCfs = baseCfs;
+        this.readCommand = readCommand;
+        this.column = column;
+        this.queryController = new QueryController(baseCfs,
+                (PartitionRangeReadCommand) readCommand,
+                DatabaseDescriptor.getRangeRpcTimeout());
+    }
+
+    /**
+     * TODO(p.boddu): We are searching and then creating the iterator. Incorporate searching in the iterator. This will
+     * save time from unneeded results (for e.g. limit 10)
+     *
+     * @param executionController
+     * @return
+     */
+    @Override
+    public UnfilteredPartitionIterator search(ReadExecutionController executionController) {
+        // TODO(p.boddu): Handle multiple expressions.
+        ByteBuffer indexValueBb = readCommand.rowFilter().getExpressions().get(0).getIndexValue();
+        String indexValue = UTF8Type.instance.compose(indexValueBb);
+        BooleanQuery query = patternToQuery(indexValue);
+        logger.debug("Searching Lucene index for column:{} with value:{} luceneQuery:{}",
+                column.name.toString(), indexValue, query);
+
+        Map<SSTableReader, List<Long>> searchResultsBySSTable =
```

Comment:
can you add some comments explaining what this does

Reply:
done.